### PR TITLE
Add More Ability For a Section within a Form

### PR DESCRIPTION
**Pull request by testusershiva**
1) Add-more property added for section.
2) This package should be published first and use it in implementer-interface.
3) Find PR for implementer-interface at https://github.com/Bahmni/implementer-interface/pull/4
4) JIRA: https://bahmni.atlassian.net/browse/BAH-740

### CODE REVIEW

File: src/helpers/ControlRecordTreeBuilder.js [hunk 1]
```diff
@@ -55,10 +59,45 @@ export const ControlRecord = new Record({
...
+      }
+      return this.set('children', updatedChildren);
+    }
+    return this;
+  },
 
   update(formFieldPath, value, errors, isRemoved) {
     if (this.formFieldPath === formFieldPath) {
-      return (Object.keys(value).length === 0 && isRemoved ? this.set('active', false) : this)
+      return (Object.keys(value).length === 0 && isRemoved ? this.set('active', false)
```
Comment (binduak) [inline src/helpers/ControlRecordTreeBuilder.js]: @shivarach @bsneha90 can we remove **isRemoved** parameter as its not being used with the latest changes

File: src/helpers/ControlRecordTreeMgr.js [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
+                this.generateNextTree(rootTree, r.formFieldPath, updatedTree.formFieldPath)
             ));
       updatedTree = updatedTree.set('children', clonedChildTree);
     }
-    const nextFormFieldPath = this.generateFormFieldPath(updatedTree);
-    return updatedTree.set('formFieldPath', nextFormFieldPath).set('value', {}).set('active', true);
+    return updatedTree.set('value', {}).set('active', true);
   }
 
   filterChildTree(updatedTree) {
```
Comment (binduak) [inline src/helpers/ControlRecordTreeMgr.js]: @shivarach can you please explain what are we doing as part of **filterChildTree** method ??

Comment (bsneha90) [reply src/helpers/ControlRecordTreeMgr.js]: We are filtering out any child created using add more.

File: src/helpers/ControlRecordTreeBuilder.js (hunk 1 above)
Comment (bsneha90) [reply src/helpers/ControlRecordTreeBuilder.js]: removed and committed

**Pull request merged**